packages/components/src/components/Calculator/CalculatorResult.tsx

```diff
@@ -0,0 +1,69 @@
+import { FC, useEffect, useMemo, useState } from "react";
+
+import { Env, SqValue } from "@quri/squiggle-lang";
+
+import { PlaygroundSettings } from "../PlaygroundSettings.js";
+import { ValueResultViewer } from "./ValueResultViewer.js";
+import {
+  InputValues,
+  SqCalculatorValueWithContext,
+  SqValueResult,
+} from "./types.js";
+import { useSavedCalculatorState } from "./useSavedCalculatorState.js";
+
+type Props = {
+  valueWithContext: SqCalculatorValueWithContext;
+  inputValues: InputValues;
+  environment: Env;
+  settings: PlaygroundSettings;
+};
+
+export const CalculatorResult: FC<Props> = ({
```

Comment:
Note: having this as a separate component is not just better for organizing the code but also should be more performant. Now we don't have to rerender all inputs when the function value is recalculated.

Reply:
Good catch